### ISSUE TITLE
Safari 11 crash if Chrome peer starts call

### ISSUE THREAD

**Issue by vincentwoo** [twilio/twilio-video.js#303]
- [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [ ] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.

The following situation appears to crash my app, which uses twilio-video:

1. Peer Chrome creates a room, and publishes tracks
2. Peer Safari joins this room, and publishes tracks. At this point they can see each other.
3. Peer Safari exits the room (by calling disconnect and stop on his own tracks).
4. Peer Safari receives an "unexpected error" and the browser hard reloads.

This is obviously not ideal. What makes it stranger, though, is if you instruct both peers to prefer the H264 codec, the problem goes away. I don't have a lot of ability to reason about this and was hoping someone could provide some pointers. I'd like to not have to force everyone to use H264 just to avoid these Safari browser crashes.

**Comment by markandrus** [comment]
Hi @vincentwoo,

Sorry you're hitting this. On the face of it, it sounds similar to https://github.com/twilio/twilio-video.js/issues/294; however, then you mention the behavior change with H.264, which makes it strange. I'll try to reproduce with a recent Safari Tech Preview.

Mark

**Comment by vincentwoo** [comment]
I can also confirm that this happens for me when using an unmodified Twilio Video Quickstart app.

**Comment by vincentwoo** [comment]
Actually, maybe the H264 thing isn't a fix - it appeared to work for a while but seems to be crashing now again.

**Comment by wesleykendall** [comment]
Hey @vincentwoo just wanted to clarify that I was able to work around this similar issue (as pointed out in https://github.com/twilio/twilio-video.js/issues/294) by doing this:

```
// Detach the Tracks from the DOM.
function detachTracks(tracks) {
  let isSafari = /Safari/.test(navigator.userAgent) && /Apple Computer/.test(navigator.vendor);
  tracks.forEach(function(track) {
    if (isSafari) {
      // Avoid detaching the track and only remove the DOM element for Safari.
      // This avoids the issue that happens because of https://github.com/twilio/twilio-video.js/issues/294
      // We are still uncertain about what side effects this problem ultimately has if particants
      // connect and reconnect multiple times.
      track._attachments.forEach(function(element) {
        element.remove();
      });
    } else {
      track.detach().forEach(function(detachedElement) {
        detachedElement.remove();
      });
    }
  });
}
```

In other words, on Safari, don't detach the tracks and instead remove them directly from the DOM. 

I don't know what kind of consequences this has for people leaving and joining the video, but this solution has worked for me on Safari without any issues

**Comment by wesleykendall** [comment]
@markandrus not sure if the above is too hacky of a solution for Safari, but I'm happy to try to contribute some sort of patch for helping deal with buggy versions of Safari if you want
